src/averageAges.js

```diff
@@ -37,7 +42,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenWithChildren = !withChildren
+    ? people.filter(woman => woman.sex === 'f')
+    : people.filter(woman =>
+      people.some(child => child.mother === woman.name));
+  const numberOfWomen = womenWithChildren.length;
+  const womenAges = womenWithChildren.map(woman => woman.died - woman.born);
+  const averageAge = womenAges.reduce((a, b) => a + b, 0) / numberOfWomen;
+
+  return averageAge;
```

Comment:
If it's possible always return value immediately 
```suggestion
  return womenAges.reduce((a, b) => a + b, 0) / numberOfWomen;
```